src/planscape/conditions/raster_utils_test.py

```diff
@@ -12,6 +14,129 @@
+    def setUp(self):
+        # Add a row for CRS 9822 to the spatial_ref_sys table, and the GeoTiff to the table.
+        with connection.cursor() as cursor:
+            query = ("insert into spatial_ref_sys(srid, proj4text) values(9822, '{}')").format(
+                settings.CRS_9822_PROJ4)
+            cursor.execute(query)
+
+        self.xorig = -2116971
+        self.yorig = 2100954
+        self.xscale = 300
+        self.yscale = -300
+
+    def test_returns_score(self):
+        geo = _create_geo(self, 0, 3, 0, 1)
+        foo_raster = _create_raster(self, 4, 4, (1, 2, 3, 4,
+                                                 5, 6, 7, 8,
+                                                 9, 10, 11, 12,
+                                                 13, 14, 15, 16))
+        _create_condition_raster(foo_raster, "foo")
+        score = compute_condition_score_from_raster(geo, "foo")
+        self.assertEqual(score, 36.0 / 8)
+
+    def test_returns_none_for_no_geo(self):
+        foo_raster = _create_raster(self, 4, 4, (1, 2, 3, 4,
+                                                 5, 6, 7, 8,
+                                                 9, 10, 11, 12,
+                                                 13, 14, 15, 16))
+        _create_condition_raster(foo_raster, "foo")
+        score = compute_condition_score_from_raster(None, "foo")
+        self.assertIsNone(score)
+
+    def test_fails_wrong_srid(self):
+        polygon = Polygon(
+            ((-120, 40),
+             (-120, 41),
+             (-121, 41),
+             (-121, 40),
+             (-120, 40)))
+        geo = MultiPolygon(polygon)
+        geo.srid = 4269
+        with self.assertRaises(Exception) as context:
+            compute_condition_score_from_raster(geo, "foo")
+        self.assertEqual(
+            str(context.exception), "geometry SRID is 4269 (expected 9822)")
+
+    def test_returns_none_for_no_intersection(self):
+        geo = _create_geo(self, 7, 10, 0, 1)
+        foo_raster = _create_raster(self, 4, 4, (1, 2, 3, 4,
+                                                 5, 6, 7, 8,
+                                                 9, 10, 11, 12,
+                                                 13, 14, 15, 16))
+        _create_condition_raster(foo_raster, "foo")
+        score = compute_condition_score_from_raster(geo, "foo")
+        self.assertIsNone(score)
+
+    def test_returns_none_for_no_raster(self):
+        geo = _create_geo(self, 0, 3, 0, 1)
+        foo_raster = _create_raster(self, 4, 4, (1, 2, 3, 4,
+                                                 5, 6, 7, 8,
+                                                 9, 10, 11, 12,
+                                                 13, 14, 15, 16))
+        _create_condition_raster(foo_raster, "foo")
+        score = compute_condition_score_from_raster(
+            geo, "nonexistent_raster_name")
+        self.assertIsNone(score)
+
+
 class MeanConditionScoresTest(TestCase):
```

Comment:
is this a separate test? if so, can we give it a more distinct name?

Reply:
Yes - done.